sklearn/tests/test_calibration.py

```diff
@@ -1077,20 +1081,43 @@ def test_sigmoid_calibration_max_abs_prediction_threshold(global_random_seed):
     assert_allclose(b2, b3, atol=atol)
 
 
-def test_float32_predict_proba(data):
+@pytest.mark.parametrize("use_sample_weight", [True, False])
+@pytest.mark.parametrize("method", ["sigmoid", "isotonic"])
+def test_float32_predict_proba(data, use_sample_weight, method):
     """Check that CalibratedClassifierCV works with float32 predict proba.
 
-    Non-regression test for gh-28245.
+    Non-regression test for gh-28245 and gh-28247.
     """
+    if use_sample_weight:
+        # Use dtype=np.float64 to check that this does not trigger an
+        # unintentional upcasting: the dtype of the base estimator should
+        # control the dtype of the final model.
```

Comment:
```suggestion
        # control the dtype of the final model. In particular, the
        # sigmoid calibrator relies on inputs (predictions and sample weights)
        # with consistent dtypes because it is written in Python. As this test
        # forces the predictions to be `float32`, we want to check that
        # `CalibratedClassifierCV` internally converts `sample_weight` to the same
        # dtype to avoid making the Cython crash.
```